src/types/yaml.ts

```diff
@@ -0,0 +1,80 @@
+interface Observation {
+  timestamp: string;
+  duration: number;
+  cpu: number;
+}
+
+interface Config {
+  sku: string;
+}
+
+interface Series {
+  observations: Observation[];
+}
+
+interface Queue {
+  config: Config;
+  series: Series;
+}
+
+interface ServerParams {
+  vendor: string;
+  region: string;
+}
+
+interface Servers {
+  config: Config;
+  params: ServerParams;
+  observations: Observation;
+}
+
+interface LoadBalancer {
+  model: string;
+  config: Config;
+  observation: Observation;
+}
+
+interface Backend {
+  model: string;
+  config: Config;
+  children: {
+    servers?: Servers;
+    queue?: Queue;
+  };
+}
+
+interface Edge {
+  'load-balancer': LoadBalancer;
+}
+
+interface Graph {
+  backend: Backend;
+  edge: Edge;
+}
+
+interface Pipeline {
+  plugin: string;
+}
+
+interface EnrichmentPipeline extends Pipeline {
+  grid_emissions_plugin: string;
+}
+
+interface NormalizationPipeline extends Pipeline {
+  impact_window: number;
+}
+
+interface AggregationPipeline extends Pipeline {
+  functional_unit: string;
+}
+
+export interface ImpactYaml {
+  name: string;
+  description: string;
+  tags: Record<string, string>;
+  config: Config;
```

Comment:
Again, unsure exactly what this will contain. There will be some content that is typeable but for instance this is where we will configure the plugins, and we will never know the parameters that plugins will have in their config so there is a level here where the config will just need  be [key]: any